lib/client/renderer.js

```diff
@@ -422,6 +485,116 @@ function init (client, d3) {
+    var linedata = [];
+    var notemplinedata = [];
+    var basalareadata = [];
+    var tempbasalareadata = [];
+    var from = chart().brush.extent()[0].getTime();
+    var to = Math.max(chart().brush.extent()[1].getTime(), new Date().getTime()) + client.forecastTime;
+    
+    var date = from;
+    var lastbasal = 0;
+    var lastdate = from;
+
+    while (date <= to) {
+      var basalvalue = profile.getTempBasal(date, client.tempbasaltreatments);
+      if (!_.isEqual(lastbasal, basalvalue)) {
+        linedata.push( { d: date, b: basalvalue.tempbasal } );
+        notemplinedata.push( { d: date, b: basalvalue.basal } );
+        if (basalvalue.treatment) {
+          tempbasalareadata.push( { d: date, b: basalvalue.tempbasal } );
+          basalareadata.push( { d: date, b: 0 } );
+        } else {
+          basalareadata.push( { d: date, b: basalvalue.tempbasal } );
+          tempbasalareadata.push( { d: date, b: 0 } );
+        }
+      }
+      lastbasal = basalvalue;
+      lastdate = date;
+      date += times.mins(1).msecs;
+    }
+    linedata.push( { d: to, b: profile.getTempBasal(to,client.tempbasaltreatments).tempbasal } );
+    notemplinedata.push( { d: to, b: profile.getTempBasal(to,client.tempbasaltreatments).basal } );
+    basalareadata.push( { d: to, b: profile.getTempBasal(to,client.tempbasaltreatments).basal } );
+    tempbasalareadata.push( { d: to, b: profile.getTempBasal(to,client.tempbasaltreatments).tempbasal } );
+    
+    chart().yScaleBasals.domain([0, d3.max(linedata, function(d) { return d.b; }) ]);
+    // update y axis domain
+//    chart().basals.select('.y')
+//      .call(chart().yAxisBasals);
+          
+    chart().basals.selectAll('g').remove();
+    chart().basals.selectAll('.basalline').remove().data(linedata);
+    chart().basals.selectAll('.notempline').remove().data(notemplinedata);
+    chart().basals.selectAll('.basalarea').remove().data(basalareadata);
+    chart().basals.selectAll('.tempbasalarea').remove().data(tempbasalareadata);
+//    chart().basals.selectAll('.tempbasaltext').remove();
+
+    var valueline = d3.svg.line()
+      .interpolate('step-after')
+      .x(function(d) { return chart().xScaleBasals(d.d); })
+      .y(function(d) { return chart().yScaleBasals(d.b); });
+
+    var area = d3.svg.area()
+      .interpolate('step-after')
+      .x(function(d) { return chart().xScaleBasals(d.d); })
+      .y0(chart().yScaleBasals(0))
+      .y1(function(d) { return chart().yScaleBasals(d.b); });
+      
+    var g = chart().basals.append('g');
+
+    g.append('path')
+      .attr('class', 'line basalline')
+      .attr('stroke', '#0099ff')
+      .attr('stroke-width', 2)
+      .attr('fill', 'none')
+      .attr('d', valueline(linedata))
+      .attr('clip-path', 'url(#clip)');
+
+    g.append('path')
+      .attr('class', 'line notempline')
+      .attr('stroke', '#0099ff')
+      .attr('stroke-width', 2)
+      .attr("stroke-dasharray", ("3, 3"))
```

Comment:
![Codacy](https://www.codacy.com/versioned/images/favicon.png) Issue found: [Mixed double and single quotes.](https://www.codacy.com/app/Nightscout/cgm-remote-monitor/file/1129533694/issues/source?bid=2452379&fileBranchId=2886508#l560)